src/res_core.ml

```diff
@@ -2604,6 +2605,28 @@ and parseJsxProp p =
           if optional then Asttypes.Optional name else Asttypes.Labelled name
         in
         Some (label, attrExpr))
+  (* {...props} *)
+  | Lbrace -> (
```

Comment:
What happens when this is parsed while V3 is active?
Should this be gated on version or is it OK like this?

Reply:
It will be OK with V3. V9 will give us a parsing error, but V10 will parse the JSX application as below, but it will be a type error anyway. Because the definition doesn't have a `spreadProps` as a prop. IMO, it doesn't make anything worse.
```rescript
// lowercase
ReactDOMRe.createDOMElementVariadic(
  "div",
  ~props=ReactDOMRe.domProps(~spreadProps=foo, ()),
  [],
)

// uppercase
React.createElement(Foo.make, Foo.makeProps(~spreadProps=foo, ())),
```